ultradns-rest/src/main/java/denominator/ultradns/UltraDNSRestProvider.java

```diff
@@ -0,0 +1,187 @@
+package denominator.ultradns;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.inject.Singleton;
+
+import dagger.Provides;
+import denominator.BasicProvider;
+import denominator.CheckConnection;
+import denominator.DNSApiManager;
+import denominator.QualifiedResourceRecordSetApi;
+import denominator.ResourceRecordSetApi;
+import denominator.ZoneApi;
+import denominator.config.ConcatBasicAndQualifiedResourceRecordSets;
+import denominator.config.NothingToClose;
+import denominator.config.WeightedUnsupported;
+import denominator.profile.GeoResourceRecordSetApi;
+//import UltraDNSRestErrorDecoder.UltraDNSError;
+import feign.Feign;
+import feign.Logger;
+import feign.Request.Options;
+import feign.form.FormEncoder;
+import feign.gson.GsonDecoder;
+import feign.gson.GsonEncoder;
+
+import static dagger.Provides.Type.SET;
+
+public class UltraDNSRestProvider extends BasicProvider {
+
+  private final String url;
+
+  public UltraDNSRestProvider() {
+    this(null);
+  }
+
+  /**
+   * @param url if empty or null use default
+   */
+  public UltraDNSRestProvider(String url) {
+    this.url =
+            url == null || url.isEmpty() ? "https://test-restapi.ultradns.com/v2" : url;
+  }
+
+  @Override
+  public String url() {
+    return url;
+  }
+
+  /**
+   * harvested from the {@code RESOURCE RECORD TYPE CODES} section of the SOAP user guide, dated
+   * 2012-11-04.
+   */
+  @Override
+  public Set<String> basicRecordTypes() {
+    Set<String> types = new LinkedHashSet<String>();
+    types.addAll(
+            Arrays.asList("A", "AAAA", "CNAME", "HINFO", "MX", "NAPTR", "NS", "PTR", "RP", "SOA", "SPF",
```

Comment:
I think its better to create an enum for all these record type constants & use that enum throughout the code.